Rubberduck.RegexAssistant/Atom.cs

```diff
@@ -0,0 +1,284 @@
+﻿using Rubberduck.RegexAssistant.i18n;
+using System;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+
+namespace Rubberduck.RegexAssistant
+{
+    public interface Atom : IDescribable
+    {
+        string Specifier { get; }
+    }
+
+    public class CharacterClass : Atom
+    {
+        public static readonly string Pattern = @"(?<!\\)\[(?<expression>.*?)(?<!\\)\]";
+        private static readonly Regex Matcher = new Regex("^" + Pattern + "$");
+
+        public bool InverseMatching { get; }
+        public IList<string> CharacterSpecifiers { get; }
+        private readonly string specifier;
+
+        public CharacterClass(string specifier)
+        {
+            Match m = Matcher.Match(specifier);
+            if (!m.Success)
+            {
+                throw new ArgumentException("The given specifier does not denote a character class");
+            }
+            this.specifier = specifier;
+            string actualSpecifier = m.Groups["expression"].Value;
+            InverseMatching = actualSpecifier.StartsWith("^");
+            CharacterSpecifiers = new List<string>();
+
+            ExtractCharacterSpecifiers(InverseMatching ? actualSpecifier.Substring(1) : actualSpecifier);
+        }
+
+        public string Specifier
+        {
+            get
+            {
+                return specifier;
+            }
+        }
+
+        private static readonly Regex CharacterRanges = new Regex(@"(\\[dDwWsS]|(\\[ntfvr]|\\([0-7]{3}|x[\dA-F]{2}|u[\dA-F]{4}|[\\\.\[\]])|.)(-(\\[ntfvr]|\\([0-7]{3}|x[A-F]{2}|u[\dA-F]{4}|[\.\\\[\]])|.))?)");
+        private void ExtractCharacterSpecifiers(string characterClass)
+        {
+            MatchCollection specifiers = CharacterRanges.Matches(characterClass);
+            
+            foreach (Match specifier in specifiers)
+            {
+                if (specifier.Value.Contains("\\"))
```

Comment:
not sure if you actually can do that here, since we want to add after processing the internal stuff (or possibly not depending on what we encounter)